app/stores/BaseStore.js

```diff
@@ -140,6 +140,11 @@ export default class BaseStore<T: BaseModel> {
     }
 
     let item = this.data.get(id);
+
+    if (this.modelName === "Collection") {
+      item = this.rootStore.collections.getByUrl(id);
+    }
+
```

Comment:
@tommoor, Is this okay, or Should I consider creating fetch in the collection store?

Reply:
Override fetch in collection store pls :) This feature is likely to mean a bit of copy/paste from documents but that's okay.